### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by sachavskyi**

### CODE REVIEW

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]
Use `@pytest.mark.parametrize`.  Read checklist

File: app/test_main.py [hunk 1]
```diff
@@ -1,3 +1,53 @@
...
+
+@pytest.mark.parametrize(
+    "cat_age,dog_age,expected_ages",
+    [
+        (14, 14, [0, 0]),
+        (23, 23, [1, 1]),
+        (15, 15, [1, 1]),
+        (24, 24, [2, 2]),
+        (28, 28, [3, 2]),
+        (100, 100, [21, 17])
```
Comment (Dimosphen1) [inline app/test_main.py:14]: Add test cases for negative numbers, zero numbers, and different numbers for cat and dog age as mentioned [in the checklist](https://github.com/sachavskyi/py-cat-and-dog-years/blob/develop/checklist.md#clean-code).

File: app/test_main.py [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+        ("5", 15, TypeError),
+        (15, [15], TypeError),
+        ({2}, 15, TypeError),
+
+    ],
+    ids=[
+        "should raise error if value is str",
+        "should raise error if value is list",
+        "should raise error if value is dict",
+    ]
```
Comment (Dimosphen1) [inline app/test_main.py:45]: There is no need for specifying `ids` here.

File: app/test_main.py [hunk 3]
```diff
@@ -1,3 +1,53 @@
...
+        (100, 100, [21, 17])
+    ],
+    ids=[
+        "if values less than 15",
+        "if values less than 24",
+        "if values equal 15",
+        "if values equal 24",
+        "if values have extra years",
+        "if values are large"
+    ]
```
Comment (Dimosphen1) [inline app/test_main.py:23]: There is no need for specifying `ids` here.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Additional test cases were requested.

File: app/test_main.py [hunk 4]
```diff
@@ -1,3 +1,43 @@
...
+        ("5", 15, TypeError),
+        (15, [15], TypeError),
+        ({2}, 15, TypeError),
+
+    ]
+)
+def test_should_return_correct_exception(
+        cat_age: Any,
+        dog_age: Any,
+        expected_error: Any
```
Comment (pavlejviki) [inline app/test_main.py:40]: wrong type annotation for expected_error

**Review by pavlejviki — CHANGES_REQUESTED** [review]

**Review by Bohdan-Salamakha — APPROVED** [review]
